login-service/src/api-login/dto/link-user.dto.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpException, HttpStatus } from "@nestjs/common";
+
+/**
+ * Input type for self-linking
+ * Only expects the registration_token
+ */
+export class RegistrationTokenInput {
+    /**
+     * The register token issued during as result of the oauth registration flow.
+     * Scope of the token must contain "login-register".
+     *
+     * Must be given.
+     *
+     * @example "registration.token.jwt"
+     */
+    register_token: string;
+
+    /**
+     * Checks that the given object contains a register_token string that in not empty
+     *
+     * @param input The instance to check
+     * @returns The argument unchanged
+     * @throws {@link HttpException} BAD_REQUEST if invalid
+     */
+    static check(input: RegistrationTokenInput): RegistrationTokenInput {
+        if (typeof input.register_token != "string" || input.register_token.trim().length <= 0) {
```

Comment:
use `== 0` for unsigned comparison

Reply:
How do you know it is unsigned?
There is no such concept in js

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#description
it is a [precise integers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) with the number of utf16 units. 0 is explicitly given and no string can be shorter than 0 utf16 units

Reply:
Didn't I fix this already?